woocommerce/woocommerce
# Extract manifest generation from WooCommerce docs plugin into a CLI tool

The WooCommerce Docs plugin is designed to consume a basic manifest format. Up till now that prototype format was generated in a script within the plugin, but its desired to separate the tool out so it can be published and consumed from anywhere to generate manifest files from Markdown docs.

This extracts the logic for that script from the plugin into monorepo utils CLI and adds some supporting tests for the logic. This also updates instructions in README files so that you can generate a manifest file yourself in development in the WooCommerce docs plugin.


### Testing Instructions


1. Ensure that pnpm install has been run on a checkout of this branch
2. Run `pnpm build` on the monorepo utils package
3. From the root you can generate a manifest into the WooCommerce docs plugin as instructed [in the README](https://github.com/woocommerce/woocommerce/pull/39161/files#diff-e6483b5a793df3870167230500ed1be3fadf62c3cf8d59d411fba3d134bc5bbe) that comes with this PR and see that the manifest is generated correctly. 
4. Also see that tests run successfully via `pnpm test`, because we're still not running this packages tests along with the rest of the repo right now (something I intend to fix soon)
